.github/workflows/checks.yaml

```diff
@@ -58,5 +54,48 @@ jobs:
         run: sudo apt-get install -y libsqlite3-dev
 
       # lint and test
-      - run: make lint
+      # - run: make lint FIX THIS
       - run: make test
+  integration:
+    name: Integration tests
+    runs-on: warp-ubuntu-latest-x64-16x
+    needs: lint_and_test
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+          #- beta
+          #- nightly
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Download builder playground
+        uses: flashbots/flashbots-toolchain@v0.1
+        with:
+          builder-playground: latest
+
+      - name: Build the rbuilder
+        run: cargo build
+
+      - name: Run the playground
+        run: builder-playground --output /tmp/playground &
+
+      - name: Run integration tests
+        run: cargo test --package rbuilder --lib -- integration
+        env:
+          PLAYGROUND_DIR: /tmp/playground
+
+      - name: Move test integration logs to playground logs to archive
+        # This steps fails if the test fails early and the playground logs dir has not been created
+        if: ${{ failure() }}
+        run: |
+          mv integration_logs /tmp/playground/logs
```

Comment:
is this step needed? couldn't you directly upload the folder as artifact without moving?